### DISCUSSION TITLE
Github Action with Suggested Changes for PRs?

### DISCUSSION THREAD

**Discussion by polarathene** [textlint/textlint#768] [Ideas]
I have been contributing to docs and often the review process might bring up common issues not only with style conventions which are more reliable to automate, but prose. 

`textlint` seems to be quite interesting with the latter, but probably better to have a Github Action that uses the Suggested Changes github feature to allow the author to conditionally apply the reported changes manually, or discuss with maintainers (with the maintainers potentially having less review suggestions to make).

Unclear how that is setup, I've seen Github Actions for running linters and automatically applying fixes without the Suggested Changes feature used. If this is something being used already, it'd be great if the docs could cover this. I assume it'd need a custom formatter to wrap each change into the Suggested Changes codefence syntax, then send that output as a PR comment/review requesting changes?

**Comment by polarathene** [comment]
Might be that the third-party support on docs with tools like [reviewdog](https://github.com/reviewdog/reviewdog) should be handling this?

**EDIT:** Found the feature in reviewdog as a recent beta development: https://github.com/reviewdog/reviewdog/issues/678

**Comment by azu** [comment]
Suggested Changes is a nice feature, but 3rd-party like ReviewDog may do it.

As baseline, [setup-node](https://github.com/actions/setup-node) support ESLint's stylish format.

- https://github.com/actions/setup-node/blob/59e61b89511ed136a0b17773f07c349fa5c01e8b/src/main.ts#L36
- [toolkit/problem-matchers.md at master · actions/toolkit](https://github.com/actions/toolkit/blob/master/docs/problem-matchers.md)

It shows a lint error message on the PR.
Probably, It also supports textlint's stylish format(default output).

I agree that we need to add documentation about CI + textlint.
It includes 3rd-party integration.

- Setup textlint in CI
- Integration with GitHub Actions
- etc...

**Comment by polarathene** [comment]
I think the reviewdog integration might be the way to go for me, I spent some time looking for how others might have been using the two Github features together but it doesn't seem that common I guess.

Once the `action-suggest` reviewdog Github Action is out of beta, it might be a nice addition in the docs, so that a user looking for usecase can do so easily without spending too much time into figuring out how (I was starting to think I'd have to write a bunch of code for it myself).

Thanks for the links, I haven't looked at source, but there is a [textlint Github Action for reviewdog](https://github.com/tsuyoshicho/action-textlint), which AFAIK should mostly be what you'd want for a generic Github Action minus the extras to support reviewdog, likewise they have a similar example linked from the reviewdog README for [Circle CI textlint](https://github.com/azu/textlint-reviewdog-example) that could be referenced.

**Comment by azu** [comment]
We might want to add `ci.md` into https://github.com/textlint/textlint/tree/master/docs as the first step.

**Comment by kimburgess** [comment]
[Orthograph-err](https://github.com/place-labs/orthograph-err) is a thin abstraction over TextLint which does exactly this. Currently it inserts annotations using the checks API, however using review comments to add suggestions for fixable items is an excellent idea.
